tasmota/xdrv_62_robotdyn.ino

```diff
@@ -0,0 +1,253 @@
+      p->pin = -1; // mark the light as processed
+      i++; // one more light  started
+      }
+    if( k > 100) break; //watchdog
+    now = rbdWait(0,0); // wait 10µS minimum
+  }
+ return; // LOG DEBUG
+  RDBTRACE *t;
+    AddLog(LOG_LEVEL_INFO, PSTR(""));
+    t = &rbdt[0]; AddLog(LOG_LEVEL_INFO, PSTR("====== RBD  4 CALC   time %8dµs / cycle  %dµs = n %3d, lastCycle = %dµS  After last interupt"),
+                                                                    t->time,     t->cycle, t->n,     t->lastCycle);
+    t = &rbdt[1]; AddLog(LOG_LEVEL_INFO, PSTR("=======RBD  5 FIND   last %dµs  next %dµs delta %d After find nearest next"),
+                                                                t-> last, t->next, t->next - t->last );
+    t = &rbdt[2]; AddLog(LOG_LEVEL_INFO, PSTR("=======RBD  6 AVG    last %dµs  next %dµs delta %d, cycle = %dµS  After average next and cycle"),
+                                                                t-> last, t->next, t->next - t->last, t->cycle );
+    t = &rbdt[3]; AddLog(LOG_LEVEL_INFO, PSTR("=======RBD  7 GET  ZC now %dµs  next %dµs lag %4d, After calc next ZC"),
+                                                                t-> now, t->next, t->now - t->next );
+    t = &rbdt[4]; AddLog(LOG_LEVEL_INFO, PSTR("=======RBD  8 WAIT ZC now %dµs  next %dµs lag %4d, After ZC"),
+                                                                t-> now, t->next, t->now - t->next );
+
+  return;
+  for(d = 5; d < 5 + MAX_PWMS;d++) {
+    t = &rbdt[d]; AddLog(LOG_LEVEL_INFO, PSTR("=======RBD %2d STOP    now %dµs start %dµs lag %d, pin %d  After stop ZC"),
+                                                               d+5, t-> now, t->start, t->now - t->start, t->pin );
+  }
+  for(d = 10; d < 10 + k; d++) {
+    t = &rbdt[d]; AddLog(LOG_LEVEL_INFO, PSTR("=======RBD %2d START   now %dµs start %dµs lag %d, pin %d, j=%2d, p=%d    After start"),
+                                                        d+10, t-> now, t->start, t->now - t->start, t->pin, t->cycle, t->time );
+  }
+  return;
+}
+
+// commande "RDBTest"
+void rbdCmndTest(void)
+{
+  AddLog(LOG_LEVEL_INFO, PSTR("========== RBD rdbCmndTest(void)"));
+   ResponseClear();
+}
+
+/*********************************************************************************************\
+ * Driver Interface
+\*********************************************************************************************/
+
+// this is the dictionnary of commands for RobotDyn
+// first table contain the list of commandes and the seconde is the fonctions to call in this module 
+const char kRBDCommands[] PROGMEM = "|RBDTest" ;
+void (* const RBDCommands[])(void) PROGMEM = {&rbdCmndTest };
+// This function is ethe interface (the entry point for all action require by tastota) with tasmota program
+
+bool Xdrv62(uint8_t function) {
+
+  bool result = false;
+  switch(function) {
+    case FUNC_INIT : // module must init
+      rbdInit();
```

Comment:
A driver needs to handle it's back off when there it is not configured by either selecting a unique GPIO or a setting. 

Your driver, once enabled in the compile, will report an error if GPIO_ZEROCROSS is not defined on any device. Not user friendly. 

It should also backoff servicing any other function call if it's not initialized correctly.